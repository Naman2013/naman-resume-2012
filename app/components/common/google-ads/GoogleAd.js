import React, { Component } from 'react';
import PropTypes from 'prop-types';

class GoogleAd extends Component {
  static propTypes = {
    adURL: PropTypes.string.isRequired,
    adWidth: PropTypes.number.isRequired,
    adHeight: PropTypes.number.isRequired,
    targetDivID: PropTypes.string.isRequired,
  };

  componentDidMount() {
    const { adURL, adWidth, adHeight, targetDivID } = this.props;

    if (window.googletag) {
      window.googletag.cmd.push(() => {
        window.googletag.defineSlot(
          adURL,
          [adWidth, adHeight],
          targetDivID)
          .addService(window.googletag.pubads());
        window.googletag.pubads().enableSingleRequest();
        window.googletag.enableServices();
        window.googletag.display(targetDivID);
      });
    }
  }

  componentWillUnmount() {
    window.googletag.destroySlots();
  }

  render() {
    const { targetDivID, adWidth, adHeight } = this.props;
    const adInlineStyle = {
      width: adWidth,
      height: adHeight,
    };

    return (
      <div className="root">
        <div className="advertisement" id={targetDivID} style={adInlineStyle} />
        <p>Advertisement</p>

        <style jsx>
          {`
            .root {
              text-align: center;
              margin-bottom: 10px;
            }

            .advertisement {
              margin: 0 auto;
            }

            p {
              font-size: 10px;
              margin-top: 5px;
            }
          `}</style>
      </div>
    );
  }
}

export default GoogleAd;
