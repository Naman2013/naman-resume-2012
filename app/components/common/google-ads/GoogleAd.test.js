import React from 'react';
import renderer from 'react-test-renderer';
import GoogleAd from './GoogleAd';

const TEST_DATA = {
  targetDivID: 'advertisement-target',
  adURL: 'slooh.com',
  adWidth: 300,
  adHeight: 300,
};

it('renders correctly', () => {
  const googleAdTree = renderer.create(
    <GoogleAd
      targetDivID={TEST_DATA.targetDivID}
      adURL={TEST_DATA.adURL}
      adWidth={TEST_DATA.adWidth}
      adHeight={TEST_DATA.adHeight}
    />,
  ).toJSON();

  expect(googleAdTree).toMatchSnapshot();
});
