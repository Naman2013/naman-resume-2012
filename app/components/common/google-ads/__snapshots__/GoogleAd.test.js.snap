// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="root"
  data-jsx={1944505430}
>
  <div
    className="advertisement"
    data-jsx={1944505430}
    id="advertisement-target"
    style={
      Object {
        "height": 250,
        "width": 300,
      }
    }
  />
  <p
    data-jsx={1944505430}
  >
    Advertisement
  </p>
  <style
    jsx={true}
  >
    
                .root {
                  text-align: center;
                }
    
                .advertisement {
                  margin: 0 auto;
                }
    
                p {
                  font-size: 10px;
                  margin-top: 10px;
                }
              
  </style>
</div>
`;
